helm/dagster/charts/dagster-user-deployments/templates/deployment-user.yaml

```diff
@@ -87,7 +87,16 @@ spec:
                 optional: true
             {{- end }}
           resources: {{ $deployment.resources | toYaml | nindent 12 }}
-        {{- if $deployment.readinessProbe }}
+        # Only disable readiness if explicitly set to false
+        {{- if not $deployment.readinessProbe }}
+          readinessProbe:
+            exec:
+              command: ["dagster", "api", "grpc-health-check", "-p", "{{ $deployment.port }}"]
+            periodSeconds: 20
+            timeoutSeconds: 10
+            successThreshold: 1
+            failureThreshold: 15
```

Comment:
i'm not sure how representative it is, but there a number of users who have generated definitions who have to extend the server startup timeout setting in cloud deployments

Reply:
I'd still expect that to be ok unless the generation time is more than 5 minutes- these settings are really lenient, but there doesn't seem to be too big a drawback to that.

15 allowed failures * one per 20 seconds = 300 seconds. Not counting time waiting on each invocation of `grpc-health-check`, which is allowed to take up to 10 seconds